packages/grafana-ui/src/components/Tags/Tag.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FC } from 'react';
+import { cx, css } from 'emotion';
+import { GrafanaTheme } from '@grafana/data';
+import { useTheme } from '../../themes';
+import { getTagColorsFromName } from '../../utils';
+
+interface TagProps {
+  /** Name of the tag to display */
+  name: string;
+  onClick?: (name: string) => any;
+  /** Custom styles for the wrapper component */
+  className?: string;
+  [key: string]: any;
+}
+
+export const Tag: FC<TagProps> = ({ name, onClick, className, ...rest }) => {
+  const theme = useTheme();
+  const styles = getTagStyles(theme, name);
+
+  const onTagClick = () => {
+    typeof onClick === 'function' && onClick(name);
```

Comment:
Never thought about checking if it's actually a function. :smiley: I like the idea! Is it enough otherwise to just check if it isn't undefined and let TypeScript handle the type?

Reply:
If you just do `if (onClick)` this will fail when it's not a function (edge case, but better be safe :D), no sure if TS gonna handle that. Another option would be to provide a noop default `onClick`, as it avoids the need for the check, which is actually maybe what I'd have done here 🤔

Reply:
won't typescript check if it' a function?

Reply:
True, probably `if (onClick)` and leaving the rest to typescript check is enough in this case. I'm just used to explicit checks from working with JS.